File: 257-binary-tree-paths/257-binary-tree-paths.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
        
    def binaryTreePaths(self, root: Optional[TreeNode]) -> List[str]:
        paths = []
        
        def helper(root, paths, currentPath):
            if not root:
                return
            
            currentPath += str(root.val) + "->"
            
            if not root.left and not root.right:
                # Strip the right 
                path = currentPath.rstrip("->")
                # Add to paths
                paths.append(path)
                
            helper(root.left, paths, currentPath)
            helper(root.right, paths, currentPath)
            
        helper(root, paths, "")
        
        return paths
        

File: 257-binary-tree-paths/257-binary-tree-paths.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void addPath(TreeNode* root, vector<string>& output, string cur) {
        if(!root || (!root->left && !root->right)) {
            if(root) {
                cur+= to_string(root->val);
            }
            
            output.push_back(cur);
            return;
        }
        
        cur+= (to_string(root->val) + "->");

        if(root->left) {
            addPath(root->left, output, cur);
        }
        
        if(root->right) {
            addPath(root->right, output, cur);
        }
    }
    
    vector<string> binaryTreePaths(TreeNode* root) {
        
        vector<string> output;
        string cur = "";
        
        addPath(root,output,cur);
        
        return output;
        
    }
};